### ISSUE TITLE
🐛 Bug Report: missing values when using backstage-cli config:docs

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#22861]
### 📜 Description

When running `yarn backstage-cli config:docs` some of the config is missing in the end results. After some investigation this looks to be when imports as used in the source `config.d.ts` file.

### 👍 Expected behavior

When running `yarn backstage-cli config:docs` all the config defined in the source `config.d.ts` file as well as any imports should be properly documented in the output.

### 👎 Actual Behavior with Screenshots

I'm using Linguist as an example for simplicity sake but this is the case with others as well.

![Screenshot 2024-02-10 at 1 34 31 PM](https://github.com/backstage/backstage/assets/67169551/e348e382-5c03-4cb5-bf0e-7caebf2fa0a2)

Notice how `lingusit.schedule` is totally empty? It should have values for it based on this:

https://github.com/backstage/backstage/blob/57397e7d6d2d725712c439f4ab93f2ac6aa27bf8/plugins/linguist-backend/config.d.ts#L24

Which imports this:

https://github.com/backstage/backstage/blob/57397e7d6d2d725712c439f4ab93f2ac6aa27bf8/packages/backend-tasks/src/tasks/types.ts#L61-L150

### 👟 Reproduction steps

1. Clone the Backstage repo
2. Run `yarn install`
3. run `yarn tsc`
4. run `yarn backstage-cli config:docs`
5. In the web page that opens navigate to the linguist section

**Notice:** the `lingusit.schedule` is totally empty

### 📃 Provide the context for the Bug.

It's really helpful to be able to see all the config that is available to an Adopter view this command, if things are missing then they don't know about options they could be using.

### 🖥️ Your Environment

This really isn't environment specific but here are the details:

```text
OS:   Darwin 23.3.0 - darwin/arm64
node: v18.18.2
yarn: 3.2.3
cli:  0.25.2-next.2 (local)
backstage:  N/A

Dependencies:
  @backstage/app-defaults                                            0.0.0-use.local
  @backstage/backend-app-api                                         0.0.0-use.local
  @backstage/backend-common                                          0.0.0-use.local
  @backstage/backend-defaults                                        0.0.0-use.local
  @backstage/backend-dev-utils                                       0.0.0-use.local
  @backstage/backend-dynamic-feature-service                         0.0.0-use.local
  @backstage/backend-openapi-utils                                   0.0.0-use.local
  @backstage/backend-plugin-api                                      0.0.0-use.local
  @backstage/backend-tasks                                           0.0.0-use.local
  @backstage/backend-test-utils                                      0.0.0-use.local
  @backstage/catalog-client                                          1.5.2, 0.0.0-use.local
  @backstage/catalog-model                                           1.4.3, 0.0.0-use.local
  @backstage/cli-common                                              0.0.0-use.local
  @backstage/cli-node                                                0.0.0-use.local
  @backstage/cli                                                     0.0.0-use.local
  @backstage/codemods                                                0.0.0-use.local
  @backstage/config-loader                                           0.0.0-use.local
  @backstage/config                                                  0.0.0-use.local
  @backstage/core-app-api                                            0.0.0-use.local
  @backstage/core-compat-api                                         0.0.0-use.local
  @backstage/core-components                                         0.13.10, 0.0.0-use.local
  @backstage/core-plugin-api                                         1.8.2, 0.0.0-use.local
  @backstage/create-app                                              0.0.0-use.local
  @backstage/dev-utils                                               0.0.0-use.local
  @backstage/e2e-test-utils                                          0.0.0-use.local
  @backstage/errors                                                  0.0.0-use.local
  @backstage/eslint-plugin                                           0.0.0-use.local
  @backstage/frontend-app-api                                        0.0.0-use.local
  @backstage/frontend-plugin-api                                     0.5.0, 0.0.0-use.local
  @backstage/frontend-test-utils                                     0.0.0-use.local
  @backstage/integration-aws-node                                    0.0.0-use.local
  @backstage/integration-react                                       1.1.23, 0.0.0-use.local
  @backstage/integration                                             1.8.0, 0.0.0-use.local
  @backstage/plugin-adr-backend                                      0.0.0-use.local
  @backstage/plugin-adr-common                                       0.0.0-use.local
  @backstage/plugin-adr                                              0.0.0-use.local
  @backstage/plugin-airbrake-backend                                 0.0.0-use.local
  @backstage/plugin-airbrake                                         0.0.0-use.local
  @backstage/plugin-allure                                           0.0.0-use.local
  @backstage/plugin-analytics-module-ga4                             0.0.0-use.local
  @backstage/plugin-analytics-module-ga                              0.0.0-use.local
  @backstage/plugin-analytics-module-newrelic-browser                0.0.0-use.local
  @backstage/plugin-apache-airflow                                   0.0.0-use.local
  @backstage/plugin-api-docs-module-protoc-gen-doc                   0.0.0-use.local
  @backstage/plugin-api-docs                                         0.0.0-use.local
  @backstage/plugin-apollo-explorer                                  0.0.0-use.local
  @backstage/plugin-app-backend                                      0.0.0-use.local
  @backstage/plugin-app-node                                         0.0.0-use.local
  @backstage/plugin-app-visualizer                                   0.0.0-use.local
  @backstage/plugin-auth-backend-module-atlassian-provider           0.0.0-use.local
  @backstage/plugin-auth-backend-module-aws-alb-provider             0.0.0-use.local
  @backstage/plugin-auth-backend-module-gcp-iap-provider             0.0.0-use.local
  @backstage/plugin-auth-backend-module-github-provider              0.0.0-use.local
  @backstage/plugin-auth-backend-module-gitlab-provider              0.0.0-use.local
  @backstage/plugin-auth-backend-module-google-provider              0.0.0-use.local
  @backstage/plugin-auth-backend-module-microsoft-provider           0.0.0-use.local
  @backstage/plugin-auth-backend-module-oauth2-provider              0.0.0-use.local
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider        0.0.0-use.local
  @backstage/plugin-auth-backend-module-oidc-provider                0.0.0-use.local
  @backstage/plugin-auth-backend-module-okta-provider                0.0.0-use.local
  @backstage/plugin-auth-backend-module-pinniped-provider            0.0.0-use.local
  @backstage/plugin-auth-backend-module-vmware-cloud-provider        0.0.0-use.local
  @backstage/plugin-auth-backend                                     0.0.0-use.local
  @backstage/plugin-auth-node                                        0.0.0-use.local
  @backstage/plugin-azure-devops-backend                             0.0.0-use.local
  @backstage/plugin-azure-devops-common                              0.0.0-use.local
  @backstage/plugin-azure-devops                                     0.0.0-use.local
  @backstage/plugin-azure-sites-backend                              0.0.0-use.local
  @backstage/plugin-azure-sites-common                               0.0.0-use.local
  @backstage/plugin-azure-sites                                      0.0.0-use.local
  @backstage/plugin-badges-backend                                   0.0.0-use.local
  @backstage/plugin-badges                                           0.0.0-use.local
  @backstage/plugin-bazaar-backend                                   0.0.0-use.local
  @backstage/plugin-bazaar                                           0.0.0-use.local
  @backstage/plugin-bitbucket-cloud-common                           0.0.0-use.local
  @backstage/plugin-bitrise                                          0.0.0-use.local
  @backstage/plugin-catalog-backend-module-aws                       0.0.0-use.local
  @backstage/plugin-catalog-backend-module-azure                     0.0.0-use.local
  @backstage/plugin-catalog-backend-module-backstage-openapi         0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-cloud           0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-server          0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gcp                       0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gerrit                    0.0.0-use.local
  @backstage/plugin-catalog-backend-module-github-org                0.0.0-use.local
  @backstage/plugin-catalog-backend-module-github                    0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gitlab                    0.0.0-use.local
  @backstage/plugin-catalog-backend-module-incremental-ingestion     0.0.0-use.local
  @backstage/plugin-catalog-backend-module-ldap                      0.0.0-use.local
  @backstage/plugin-catalog-backend-module-msgraph                   0.0.0-use.local
  @backstage/plugin-catalog-backend-module-openapi                   0.0.0-use.local
  @backstage/plugin-catalog-backend-module-puppetdb                  0.0.0-use.local
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model   0.0.0-use.local
  @backstage/plugin-catalog-backend-module-unprocessed               0.0.0-use.local
  @backstage/plugin-catalog-backend                                  0.0.0-use.local
  @backstage/plugin-catalog-common                                   1.0.20, 0.0.0-use.local
  @backstage/plugin-catalog-graph                                    0.0.0-use.local
  @backstage/plugin-catalog-import                                   0.0.0-use.local
  @backstage/plugin-catalog-node                                     0.0.0-use.local
  @backstage/plugin-catalog-react                                    1.9.3, 0.0.0-use.local
  @backstage/plugin-catalog-unprocessed-entities                     0.0.0-use.local
  @backstage/plugin-catalog                                          0.0.0-use.local
  @backstage/plugin-cicd-statistics-module-gitlab                    0.0.0-use.local
  @backstage/plugin-cicd-statistics                                  0.0.0-use.local
  @backstage/plugin-circleci                                         0.0.0-use.local
  @backstage/plugin-cloudbuild                                       0.0.0-use.local
  @backstage/plugin-code-climate                                     0.0.0-use.local
  @backstage/plugin-code-coverage-backend                            0.0.0-use.local
  @backstage/plugin-code-coverage                                    0.0.0-use.local
  @backstage/plugin-codescene                                        0.0.0-use.local
  @backstage/plugin-config-schema                                    0.0.0-use.local
  @backstage/plugin-cost-insights-common                             0.0.0-use.local
  @backstage/plugin-cost-insights                                    0.0.0-use.local
  @backstage/plugin-devtools-backend                                 0.0.0-use.local
  @backstage/plugin-devtools-common                                  0.0.0-use.local
  @backstage/plugin-devtools                                         0.0.0-use.local
  @backstage/plugin-dynatrace                                        0.0.0-use.local
  @backstage/plugin-entity-feedback-backend                          0.0.0-use.local
  @backstage/plugin-entity-feedback-common                           0.0.0-use.local
  @backstage/plugin-entity-feedback                                  0.0.0-use.local
  @backstage/plugin-entity-validation                                0.0.0-use.local
  @backstage/plugin-events-backend-module-aws-sqs                    0.0.0-use.local
  @backstage/plugin-events-backend-module-azure                      0.0.0-use.local
  @backstage/plugin-events-backend-module-bitbucket-cloud            0.0.0-use.local
  @backstage/plugin-events-backend-module-gerrit                     0.0.0-use.local
  @backstage/plugin-events-backend-module-github                     0.0.0-use.local
  @backstage/plugin-events-backend-module-gitlab                     0.0.0-use.local
  @backstage/plugin-events-backend-test-utils                        0.0.0-use.local
  @backstage/plugin-events-backend                                   0.0.0-use.local
  @backstage/plugin-events-node                                      0.0.0-use.local
  @backstage/plugin-explore-backend                                  0.0.0-use.local
  @backstage/plugin-explore-common                                   0.0.0-use.local
  @backstage/plugin-explore-react                                    0.0.0-use.local
  @backstage/plugin-explore                                          0.0.0-use.local
  @backstage/plugin-firehydrant                                      0.0.0-use.local
  @backstage/plugin-fossa                                            0.0.0-use.local
  @backstage/plugin-gcalendar                                        0.0.0-use.local
  @backstage/plugin-gcp-projects                                     0.0.0-use.local
  @backstage/plugin-git-release-manager                              0.0.0-use.local
  @backstage/plugin-github-actions                                   0.0.0-use.local
  @backstage/plugin-github-deployments                               0.0.0-use.local
  @backstage/plugin-github-issues                                    0.0.0-use.local
  @backstage/plugin-github-pull-requests-board                       0.0.0-use.local
  @backstage/plugin-gitops-profiles                                  0.0.0-use.local
  @backstage/plugin-gocd                                             0.0.0-use.local
  @backstage/plugin-graphiql                                         0.0.0-use.local
  @backstage/plugin-graphql-voyager                                  0.0.0-use.local
  @backstage/plugin-home-react                                       0.1.7, 0.0.0-use.local
  @backstage/plugin-home                                             0.0.0-use.local
  @backstage/plugin-ilert                                            0.0.0-use.local
  @backstage/plugin-jenkins-backend                                  0.0.0-use.local
  @backstage/plugin-jenkins-common                                   0.0.0-use.local
  @backstage/plugin-jenkins                                          0.0.0-use.local
  @backstage/plugin-kafka-backend                                    0.0.0-use.local
  @backstage/plugin-kafka                                            0.0.0-use.local
  @backstage/plugin-kubernetes-backend                               0.0.0-use.local
  @backstage/plugin-kubernetes-cluster                               0.0.0-use.local
  @backstage/plugin-kubernetes-common                                0.0.0-use.local
  @backstage/plugin-kubernetes-node                                  0.0.0-use.local
  @backstage/plugin-kubernetes-react                                 0.0.0-use.local
  @backstage/plugin-kubernetes                                       0.0.0-use.local
  @backstage/plugin-lighthouse-backend                               0.0.0-use.local
  @backstage/plugin-lighthouse-common                                0.0.0-use.local
  @backstage/plugin-lighthouse                                       0.0.0-use.local
  @backstage/plugin-linguist-backend                                 0.0.0-use.local
  @backstage/plugin-linguist-common                                  0.0.0-use.local
  @backstage/plugin-linguist                                         0.0.0-use.local
  @backstage/plugin-microsoft-calendar                               0.0.0-use.local
  @backstage/plugin-newrelic-dashboard                               0.0.0-use.local
  @backstage/plugin-newrelic                                         0.0.0-use.local
  @backstage/plugin-nomad-backend                                    0.0.0-use.local
  @backstage/plugin-nomad                                            0.0.0-use.local
  @backstage/plugin-notifications-backend                            0.0.0-use.local
  @backstage/plugin-notifications-common                             0.0.0-use.local
  @backstage/plugin-notifications-node                               0.0.0-use.local
  @backstage/plugin-notifications                                    0.0.0-use.local
  @backstage/plugin-octopus-deploy                                   0.0.0-use.local
  @backstage/plugin-opencost                                         0.0.0-use.local
  @backstage/plugin-org-react                                        0.0.0-use.local
  @backstage/plugin-org                                              0.0.0-use.local
  @backstage/plugin-pagerduty                                        0.0.0-use.local
  @backstage/plugin-periskop-backend                                 0.0.0-use.local
  @backstage/plugin-periskop                                         0.0.0-use.local
  @backstage/plugin-permission-backend-module-allow-all-policy       0.0.0-use.local
  @backstage/plugin-permission-backend                               0.0.0-use.local
  @backstage/plugin-permission-common                                0.0.0-use.local
  @backstage/plugin-permission-node                                  0.0.0-use.local
  @backstage/plugin-permission-react                                 0.4.19, 0.0.0-use.local
  @backstage/plugin-playlist-backend                                 0.0.0-use.local
  @backstage/plugin-playlist-common                                  0.0.0-use.local
  @backstage/plugin-playlist                                         0.0.0-use.local
  @backstage/plugin-proxy-backend                                    0.0.0-use.local
  @backstage/plugin-puppetdb                                         0.0.0-use.local
  @backstage/plugin-rollbar-backend                                  0.0.0-use.local
  @backstage/plugin-rollbar                                          0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-azure                  0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud        0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-bitbucket-server       0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-bitbucket              0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-confluence-to-markdown 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-cookiecutter           0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-gerrit                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-github                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-gitlab                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-rails                  0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-sentry                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-yeoman                 0.0.0-use.local
  @backstage/plugin-scaffolder-backend                               0.0.0-use.local
  @backstage/plugin-scaffolder-common                                0.0.0-use.local
  @backstage/plugin-scaffolder-node                                  0.0.0-use.local
  @backstage/plugin-scaffolder-react                                 0.0.0-use.local
  @backstage/plugin-scaffolder                                       0.0.0-use.local
  @backstage/plugin-search-backend-module-catalog                    0.0.0-use.local
  @backstage/plugin-search-backend-module-elasticsearch              0.0.0-use.local
  @backstage/plugin-search-backend-module-explore                    0.0.0-use.local
  @backstage/plugin-search-backend-module-pg                         0.0.0-use.local
  @backstage/plugin-search-backend-module-stack-overflow-collator    0.0.0-use.local
  @backstage/plugin-search-backend-module-techdocs                   0.0.0-use.local
  @backstage/plugin-search-backend-node                              0.0.0-use.local
  @backstage/plugin-search-backend                                   0.0.0-use.local
  @backstage/plugin-search-common                                    0.0.0-use.local
  @backstage/plugin-search-react                                     0.0.0-use.local
  @backstage/plugin-search                                           0.0.0-use.local
  @backstage/plugin-sentry                                           0.0.0-use.local
  @backstage/plugin-shortcuts                                        0.0.0-use.local
  @backstage/plugin-signals-backend                                  0.0.0-use.local
  @backstage/plugin-signals-node                                     0.0.0-use.local
  @backstage/plugin-signals-react                                    0.0.0-use.local
  @backstage/plugin-signals                                          0.0.0-use.local
  @backstage/plugin-sonarqube-backend                                0.0.0-use.local
  @backstage/plugin-sonarqube-react                                  0.0.0-use.local
  @backstage/plugin-sonarqube                                        0.0.0-use.local
  @backstage/plugin-splunk-on-call                                   0.0.0-use.local
  @backstage/plugin-stack-overflow-backend                           0.0.0-use.local
  @backstage/plugin-stack-overflow                                   0.0.0-use.local
  @backstage/plugin-stackstorm                                       0.0.0-use.local
  @backstage/plugin-tech-insights-backend-module-jsonfc              0.0.0-use.local
  @backstage/plugin-tech-insights-backend                            0.0.0-use.local
  @backstage/plugin-tech-insights-common                             0.0.0-use.local
  @backstage/plugin-tech-insights-node                               0.0.0-use.local
  @backstage/plugin-tech-insights                                    0.0.0-use.local
  @backstage/plugin-tech-radar                                       0.0.0-use.local
  @backstage/plugin-techdocs-addons-test-utils                       0.0.0-use.local
  @backstage/plugin-techdocs-backend                                 0.0.0-use.local
  @backstage/plugin-techdocs-module-addons-contrib                   0.0.0-use.local
  @backstage/plugin-techdocs-node                                    0.0.0-use.local
  @backstage/plugin-techdocs-react                                   0.0.0-use.local
  @backstage/plugin-techdocs                                         0.0.0-use.local
  @backstage/plugin-todo-backend                                     0.0.0-use.local
  @backstage/plugin-todo                                             0.0.0-use.local
  @backstage/plugin-user-settings-backend                            0.0.0-use.local
  @backstage/plugin-user-settings                                    0.0.0-use.local
  @backstage/plugin-vault-backend                                    0.0.0-use.local
  @backstage/plugin-vault-node                                       0.0.0-use.local
  @backstage/plugin-vault                                            0.0.0-use.local
  @backstage/plugin-xcmetrics                                        0.0.0-use.local
  @backstage/release-manifests                                       0.0.0-use.local
  @backstage/repo-tools                                              0.0.0-use.local
  @backstage/test-utils                                              0.0.0-use.local
  @backstage/theme                                                   0.4.4, 0.5.0, 0.0.0-use.local
  @backstage/types                                                   0.0.0-use.local
  @backstage/version-bridge                                          0.0.0-use.local
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by awanlin** [comment]
@Rugvip we talked about this a bit, just wanted to have something logged 👍

**Comment by awanlin** [comment]
@sennyeya, just curious if you have any thoughts on this one? You've gotten deep into the config before. No rush, when you have time. Also not looking for you to fix it, unless you want to ;)

**Comment by sennyeya** [comment]
@ahhhndre This seems to be the same situation as #20424, something to do with TS imports during the config building. I took a look at this a little while back and it was a bit of a rabbit hole. The library we're using for this generation is half-deprecated (https://github.com/YousefED/typescript-json-schema/issues/295) and we might want to switch to https://github.com/vega/ts-json-schema-generator before starting this. That new library has [caching support](https://github.com/vega/ts-json-schema-generator/pull/124) and some efficiency updates, but I haven't set up a full example so not sure if they would help us all that much.

**Comment by awanlin** [comment]
Yes, I think that's the same issue @Rugvip pointed my way. Thanks for sharing your thought as well. This is a space I'm totally lost, but I know the features `config:docs` and `config:check` are super helpful so having them work well is important.

**Comment by sennyeya** [comment]
@ahhhndre Did a bit of work towards this in https://github.com/sennyeya/backstage/tree/update-to-vega-ts-json-schema. It runs in about 8 seconds on my mac with the new library and full typescript resolution versus 4.5 seconds currently and 10.5 seconds (and an error) using the old library and `noResolve: false`. 

If we decide to go the way of the new library or just adding the references, we'll also need to update the `config.backstage.io` side to either dereference the received schema or support references when building out the page.

**Comment by Rugvip** [comment]
Sorry gonna close this as a duplicate of #20424, it's the same underlying issue so let's focus the discussion there.

@sennyeya the current implementation is too slow and we need to do something about it, so any increase at all isn't something we can take on att this point. The config schema is loaded every time the backend is started, including every restart during development. Let's discuss that in the other issue though.

**Comment by awanlin** [comment]
Works for me, just wanted to highlight this issue 👍
